examples/mass_delete_old_medias.py

```diff
@@ -0,0 +1,460 @@
+            f'viewed {media["views_last_year"]} times last year, '
+            f'{media["views_last_month"]} times last month</li>'
+            for media in medias
+        )
+        html = html_template.format(list_of_media=html_media_list, **context)
+        message.attach(MIMEText(html, 'html'))
+    return message.as_string()
+
+
+def _warn_speakers_about_upcoming_deletion(
+    msc: MediaServerClient,
+    medias: list[dict],
+    delete_date: date,
+    skip_category: str,
+    html_email_template: Path,
+    plain_email_template: Path,
+    email_subject_template: str,
+    fallback_email: str,
+    sender_address: str,
+    smtp_server: str,
+    apply: bool = False,
+):
+    try:
+        html_template = html_email_template.read_text()
+        html_is_custom = True
+    except FileNotFoundError:
+        html_template = DEFAULT_HTML_EMAIL_TEMPLATE
+        html_is_custom = False
+    try:
+        plain_template = plain_email_template.read_text()
+        plain_is_custom = True
+    except FileNotFoundError:
+        plain_template = DEFAULT_PLAIN_EMAIL_TEMPLATE
+        plain_is_custom = False
+
+    # If only one custom template is given,
+    # prioritize it and don't send the default version for the other one.
+    if html_is_custom and not plain_is_custom:
+        plain_template = None
+        logger.info(
+            'HTML email template was found but plain email template was not. '
+            'Using HTML version only.'
+        )
+    elif not html_is_custom and plain_is_custom:
+        html_template = None
+        logger.info(
+            'Plain email template was found but HTML email template was not. '
+            'Using plain version only.'
+        )
+    elif not html_is_custom and not plain_is_custom:
+        logger.info(
+            'Neither HTML nor plain email template were found. '
+            'Using default email templates.'
+        )
+    else:
+        logger.info(
+            'HTML and plain email template were found. '
+            'Using custom email templates.'
+        )
+
+    medias_per_speaker = {}
+    to_fallback = []
+    for media in medias:
+        if not media.get('speaker_email').strip():
+            to_fallback.append(media)
+            continue
+        speakers_emails = (media.get('speaker_email') or '').split('|')
+        for speaker_email in speakers_emails:
+            medias_per_speaker.setdefault(speaker_email.strip(), []).append(media)
+
+    to_send = {
+        speaker_email: _prepare_mail(
+            msc,
+            sender_address,
+            speaker_email,
+            speaker_medias,
+            delete_date,
+            skip_category,
+            html_template,
+            plain_template,
+            email_subject_template,
+        ) for speaker_email, speaker_medias in medias_per_speaker.items()
+    }
+
+    if apply:
+        password = getpass.getpass(f'Type the password for "{sender_address}" and press enter: ')
```

Comment:
I suggest we read SMTP related configuration from the config file for scriptability

Reply:
ça n'a pas de sens de lire le pwd depuis un fichier de config et de laisser le `sender_address` et `smtp_server` configurables via les options du script. Soit tout passe dans la conf, soit le pwd doit être demandé à l'exécution.

Reply:
I agree, the 3 settings must be either in the arguments or in the configuration file.

Reply:
Absolutely, sorry if that wasn't clear enough that's what i meant